### PR TITLE
BREAKING: Json signature verifications returns false on invalid json

### PR DESCRIPTION
**Pull request by lumberj**
Returning false avoids having to wrap the verify call in a try/catch everywhere it's used. For example, if a Connector receives an unsigned notification

### CODE REVIEW

**Comment by lumberj** [comment]
cc @naoitoi @MatthewPhinney @gip

**Comment by MatthewPhinney** [comment]
I think this generally looks good, but we conflate "json without signature" and "json with invalid signature" (since both return `false`). I think these two cases are important to distinguish between. A clearer API might be something like
```
verify = function (json, signature, pubKey)

**Comment by clark800** [comment]
LGTM

**Comment by lumberj** [comment]
@clark800 @MatthewPhinney I updated this so that `verify` returns an object with some information rather than true/false. The problem is the underlying libraries throw exceptions on anything that's invalid (include invalid public key or public key mismatch) which isn't useful to a caller of this method

**Comment by lumberj** [comment]
> but we conflate "json without signature" and "json with invalid signature" (since both return false)

The caller shouldn't have to care about the shape of the object or know that a signed JSON object contains a signature at the top level key `signature`. Imagine if we wanted to change from JCS to some other signing implementation that created a signed object with a different shape? The caller code would have to change.

**Comment by lumberj** [comment]
There's some duplication in the unit tests, but I'd prefer to clean that up in a later commit

**Comment by lumberj** [comment]
@clark800 can you look at the changes from the latest commit?

File: utils/jsonSigning.js [hunk 1]
```diff
@@ -109,13 +109,32 @@ function signRSA (json, prvKey) {
...
+    } else {
+      valid = false
+      error = 'Unsupported crypto algorithm'
+    }
+  } catch (e) {
+    valid = false
+    error = e.message
+  }
+
+  return {valid: valid, error: error}
```
Comment (clark800) [inline utils/jsonSigning.js]: Should we stick with `isValid` for consistency?

Comment (lumberj) [inline utils/jsonSigning.js]: Consistency with what?

Comment (clark800) [inline utils/jsonSigning.js]: I think we've generally been using the 'is' prefix for boolean names

**Comment by clark800** [comment]
latest changes LGTM

**Comment by MatthewPhinney** [comment]
> The caller shouldn't have to care about the shape of the object or know that a signed JSON object contains a signature at the top level key signature.

I wasn't trying to argue that the caller should need to understand how the json is structured, but rather that we may want more granular error messages, in the case of failed validations. Looks like 4c28bd0c149a03fc18adf52d0f7e148ee5661173 addresses this, so LGTM.

**Pull request merged**